### PR TITLE
Add Banner Component

### LINKED ISSUE #1174: Banner component
Estimated Delivery - May 31

- [x] Research
- [x] Specification
- [x] Component and tests implementation
- [x] API Documentation
- [x] Design implementation

### PR DESCRIPTION
**Pull request by RivaIvanova**
Closes #1174

### CODE REVIEW

File: src/components/banner/banner.spec.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
+import {
+  elementUpdated,
+  expect,
+  fixture,
+  html,
+  unsafeStatic,
+} from '@open-wc/testing';
+import { spy } from 'sinon';
+
+import { IgcBannerComponent, defineComponents } from '../../index.js';
```
Comment (rkaraivanov) [inline src/components/banner/banner.spec.ts:10]:
```suggestion
import { defineComponents } from '../common/definitions/defineComponents.js';
import IgcBannerComponent from './banner.js';
```

Nitpick but there is no need to import these from the top barrel index

File: src/components/banner/banner.spec.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
+import {
+  elementUpdated,
+  expect,
+  fixture,
+  html,
+  unsafeStatic,
```
Comment (rkaraivanov) [inline src/components/banner/banner.spec.ts:6]: Nit: `unsafeStatic` is rarely needed when testing a concrete component. You can template with the default `html` function.

File: src/components/banner/banner.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+export default class IgcBannerComponent extends EventEmitterMixin<
+  IgcBannerComponentEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-banner';
+  public static styles = [styles];
+
+  public static register() {
+    registerComponent(IgcBannerComponent, IgcButtonComponent);
+  }
```
Comment (rkaraivanov) [inline src/components/banner/banner.ts:50]:
```suggestion
  /* blazorSuppress */
  public static register() {
    registerComponent(IgcBannerComponent, IgcButtonComponent);
  }
```
Got to make the blazor code generation happy.

File: src/components/banner/banner.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      this._animationPlayer.play(animation()),
+    ]);
+
+    return event.type === 'finish';
+  }
+
+  private handleClick() {
+    const allowed = this.emitEvent('igcClosing', { cancelable: true });
+
+    if (allowed) {
```
Comment (rkaraivanov) [inline src/components/banner/banner.ts:110]:
```suggestion
if (this.emitEvent('igcClosing', { cancelable: true })) {
```
Nit: You can shortcut that without the extra variable.

**Review by rkaraivanov — COMMENTED** [review]
Minor comments otherwise LGTM.

However, can you fix both the CI and the storybook builds as of now I cannot really "play" with the component?

**Comment by AnjiManova** [comment]
**Material, Fluent and Bootstrap**

Paddings are fine -> 8px (left and right) to the igx-banner, 8px (left and right) to the igx-banner _message and 16px top and bottom.

**Review by rkaraivanov — APPROVED** [review]

**Pull request merged**